modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,703 @@
+        write_pro_start <- req_start
+        write_pro_end <- req_end
+        stage_get_data <- TRUE
+        raw_merge <- FALSE
+        existing_raw_file_path = NULL
+        stage_process_data <- TRUE
+        pro_merge <- FALSE
+        existing_pro_file_path = NULL
+        flag <- 1
+      }
+    } else if (nrow(raw_check <-
+                    PEcAn.DB::db.query(
+                      sprintf(
+                        "SELECT inputs.id, inputs.site_id, inputs.name, inputs.start_date, inputs.end_date, dbfiles.file_path FROM inputs INNER JOIN dbfiles ON inputs.name=dbfiles.file_name AND inputs.name LIKE '%s%%';",
+                        raw_file_name
+                      ),
+                      dbcon
+                    )) == 1) {
+      # if only raw data is requested
+      datalist <-
+        set_stage(raw_check, req_start, req_end, stage_get_data)
+      start <- as.character(datalist[[1]])
+      end <- as.character(datalist[[2]])
+      stage_get_data <- datalist[[3]]
+      raw_merge <- datalist[[4]]
+      write_raw_start <- datalist[[5]]
+      write_raw_end <- datalist[[6]]
+      stage_process_data <- FALSE
+      if(as.character(write_raw_start) == "dont write" && as.character(write_raw_end) == "dont write"){
+        raw_id <- raw_check$id
+        raw_path <- raw_check$file_path
+      }
+      if (raw_merge == TRUE) {
+        existing_raw_file_path <- raw_check$file_path
+      } else{
+        existing_raw_file_path <- NULL
+      }
+      existing_pro_file_path <- NULL
+    } else{
+      # no data of requested type exists
+      PEcAn.logger::logger.info("Requested data does not exist in the DB, retrieving for the first time")
+      flag <- 1
+      start <- req_start
+      end <- req_end
+      if (!is.null(out_get_data)) {
+        stage_get_data <- TRUE
+        write_raw_start <- req_start
+        write_raw_end <- req_end
+        raw_merge <- FALSE
+        existing_raw_file_path = NULL
+      }
+      if (!is.null(out_process_data)) {
+        stage_process_data <- TRUE
+        write_pro_start <- req_start
+        write_pro_end <- req_end
+        pro_merge <- FALSE
+        process_file_name <- NULL
+        existing_pro_file_path <- NULL
+        flag <- 1
+      }
+    }
+    
+  } else{
+    # db is completely empty for the given siteid
+    PEcAn.logger::logger.info("DB is completely empty for this site")
+    flag <- 1
+    start <- req_start
+    end <- req_end
+    stage_get_data <- TRUE
+    write_raw_start <- req_start
+    write_raw_end <- req_end
+    raw_merge <- FALSE
+    existing_raw_file_path = NULL
+    if (!is.null(out_process_data)) {
+      stage_process_data <- TRUE
+      write_pro_start <- req_start
+      write_pro_end <- req_end
+      pro_merge <- FALSE
+      existing_pro_file_path <- NULL
+    }
+  }
+  
+  outdir = file.path(outdir, paste(source, "site", siteid_short, sep = "_"))
+  
+  # call remote_process
+  output = RpTools$rp_control(
```

Comment:
before this could you do something like
```
fcn.args <- list()
fcn.args$coords     <- coords
fcn.args$outdir     <- outdir
fcn.args$start      <- start
fcn.args$end        <- end
fcn.args$source     <- source
fcn.args$collection <- collection
fcn.args$siteid     <- siteid
fcn.args$scale      <- as.double(scale)
fcn.args$projection <- projection
fcn.args$qc         <- as.double(qc)
fcn.args$algorithm              <- algorithm
fcn.args$input_file             <- input_file
fcn.args$credfile               <- credfile
fcn.args$out_get_data           <- out_get_data
fcn.args$out_process_data       <- out_process_data
fcn.args$stage_get_data         <- stage_get_data
fcn.args$stage_process_data     <- stage_process_data
fcn.args$raw_merge              <- raw_merge
fcn.args$pro_merge              <- pro_merge
fcn.args$existing_raw_file_path <- existing_raw_file_path
fcn.args$existing_pro_file_path <- existing_pro_file_path


arg.string <- listToArgString(fcn.args)

cmdFcn <- paste0("RpTools$rp_control(", arg.string, ")")
PEcAn.logger::logger.debug(paste0("Remote module executing the following function:\n", cmdFcn))
```
It helps a lot with debugging